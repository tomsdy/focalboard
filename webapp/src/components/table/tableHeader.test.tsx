// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react'
import {render} from '@testing-library/react'
import '@testing-library/jest-dom'
import {IntlProvider} from 'react-intl'

import 'isomorphic-fetch'
import {DndProvider} from 'react-dnd'
import {HTML5Backend} from 'react-dnd-html5-backend'

import {TestBlockFactory} from '../../test/testBlockFactory'

import TableHeader from './tableHeader'

const wrapProviders = (children: any) => {
    return (
        <DndProvider backend={HTML5Backend}>
            <IntlProvider locale='en'>{children}</IntlProvider>
        </DndProvider>
    )
}

describe('components/table/TableHeaderMenu', () => {
    const board = TestBlockFactory.createBoard()
    const view = TestBlockFactory.createBoardView(board)

    const view2 = TestBlockFactory.createBoardView(board)
    view2.fields.sortOptions = []

    test('should match snapshot, title column', async () => {
        const onAutoSizeColumn = jest.fn()
        const component = wrapProviders(
            <TableHeader
                readonly={false}
                sorted={'none'}
                name={'my Name'}
                board={board}
                activeView={view}
                cards={[]}
                views={[view, view2]}
                template={board.fields.cardProperties[0]}
                offset={0}
                onDrop={jest.fn()}
                onAutoSizeColumn={onAutoSizeColumn}
            />,
        )

        const {container} = render(component)
        expect(container).toMatchSnapshot()
    })
})
