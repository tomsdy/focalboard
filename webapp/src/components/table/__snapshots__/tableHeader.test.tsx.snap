// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/table/TableHeaderMenu should match snapshot, title column 1`] = `
<div>
  <div
    class="octo-table-cell header-cell"
    draggable="true"
    style="overflow: unset; width: 100px; opacity: 1;"
  >
    <div
      class="MenuWrapper"
    >
      <span
        class="Label empty "
      >
        my Name
      </span>
    </div>
    <div
      class="octo-spacer"
    />
    <div
      class="HorizontalGrip"
      draggable="true"
    />
  </div>
</div>
`;
